### PR TITLE
Adds notification for students who are removed from a course.

### PR DESCRIPTION
**Pull request by alexweinfurter**
## Description:
Students should be notified, if the teacher removes them from the course.

Closes #

## Improvements
- Notifications for the students
- Refactoring of the NotificationController
- Better reusability of the NotificationController

## Known Issues:
_NONE_

------
_Remember to prefix your PR-Title with `⚠ WIP: ` if you still work on it.
If you have reached a final state, remove the prefix._

### CODE REVIEW

File: api/package-lock.json [hunk 1]
```diff
@@ -1700,7 +1700,7 @@
     },
     "capture-stack-trace": {
       "version": "1.0.0",
-      "resolved": "https://registry.npmjs.org/capture-stack-trace/-/capture-stack-trace-1.0.0.tgz",
+      "resolved": "http://registry.npmjs.org/capture-stack-trace/-/capture-stack-trace-1.0.0.tgz",
```
Comment (aeimer) [inline api/package-lock.json]: Please let the package-lock urls as https

File: api/package-lock.json [hunk 2]
```diff
@@ -2145,7 +2145,7 @@
     },
     "create-error-class": {
       "version": "3.0.2",
-      "resolved": "https://registry.npmjs.org/create-error-class/-/create-error-class-3.0.2.tgz",
```
Comment (aeimer) [inline api/package-lock.json]: This too

File: api/package-lock.json [hunk 3]
```diff
@@ -3449,7 +3449,7 @@
     },
     "gulp-plumber": {
       "version": "1.1.0",
-      "resolved": "https://registry.npmjs.org/gulp-plumber/-/gulp-plumber-1.1.0.tgz",
```
Comment (aeimer) [inline api/package-lock.json]: And this also, thx :)

Comment (alexweinfurter) [reply api/package-lock.json]: done... why are these "resolved" lines changed? i haven't changed anything in the package-lock manually...

Comment (aeimer) [reply api/package-lock.json]: yeah, it depends on your npm version a `npm install -g npm` may help

**Pull request merged**